test/e2e/e2e.go

```diff
@@ -140,7 +140,11 @@ func waitForActivatorEndpoints(resources *v1test.ResourceObjects, clients *test.
 
 		// The subset is set. But in theory it might be revision pods,
 		// so verify below that at least 1 address is an activator pod.
-		if presources.ReadyAddressCount(svcEps) != int(sks.Spec.NumActivators) {
+		expectedActivators := presources.ReadyAddressCount(aeps)
+		if expectedActivators > int(sks.Spec.NumActivators) {
+			expectedActivators = int(sks.Spec.NumActivators)
+		}
+		if presources.ReadyAddressCount(svcEps) != expectedActivators {
```

Comment:
For me this is hard to read logically,
I'd go with
```suggestion
		wantAct := sks.Spec.NumActivators
		if numAct := presources.ReadyAddressCount(aeps); wantAct > numAct {
			wantAct. = numAct
		}
		if presources.ReadyAddressCount(svcEps) != wantAct {
```
modulo casting where necessary.

Reply:
Yes, that is better. Updated.